nevergrad/functions/images/core.py

```diff
@@ -29,34 +32,102 @@ def __init__(self, problem_name: str = "recovering", index: int = 0) -> None:
+
         super().__init__(self._loss, array)
         self.register_initialization(problem_name=problem_name, index=index)
         self._descriptors.update(problem_name=problem_name, index=index)
 
     def _loss(self, x: np.ndarray) -> float:
+        assert self.problem_name == "recovering"
         x = np.array(x, copy=False).ravel()
         x = x.reshape(self.domain_shape)
         assert x.shape == self.domain_shape, f"Shape = {x.shape} vs {self.domain_shape}"
-
         # Define the loss, in case of recovering: the goal is to find the target image.
-        assert self.problem_name == "recovering"
         assert self.index == 0
-        value = np.sum(np.fabs(np.subtract(x, self.data)))
-
-        # Here we should implement "adversarial" and others.
+        value = float(np.sum(np.fabs(np.subtract(x, self.data))))
         return value
+
+
+class TestClassifier(nn.Module):
+    def __init__(self, image_size: int = 224):
+        super().__init__()
+        self.model = nn.Linear(image_size * image_size * 3, 10)
+
+    def forward(self, x):
+        return self.model(x.flatten(x.shape[0], -1))
+
+
+# pylint: disable=too-many-arguments
+class ImageAdversarial(base.ExperimentFunction):
+
+    def __init__(self, classifier: nn.Module, image: torch.Tensor, label: int = 0, targeted: bool = False,
+                 epsilon: float = 0.05) -> None:
+        # TODO add crossover params in args + criterion
+        """
+        params : needs to be detailed
+        """
+        self.targeted = targeted
+        self.epsilon = epsilon
+        self.image = image  # if (image is not None) else torch.rand((3, 224, 224))
+        self.label = torch.Tensor([label])  # if (label is not None) else torch.Tensor([0])
+        self.label = self.label.long()
+        self.classifier = classifier  # if (classifier is not None) else Classifier()
+        self.criterion = nn.CrossEntropyLoss()
+        self.imsize = self.image.shape[1]
+
+        array = ng.p.Array(init=np.zeros(self.image.shape), mutable_sigma=True, ).set_name("")
+        array.set_mutation(sigma=self.epsilon / 10)
+        array.set_bounds(lower=-self.epsilon, upper=self.epsilon, method="clipping", full_range_sampling=True)
+        max_size = ng.p.Scalar(lower=1, upper=200).set_integer_casting()
+        array.set_recombination(ng.p.mutation.Crossover(axis=(1, 2), max_size=max_size))  # type: ignore
+
+        super().__init__(self._loss, array)
+        self.register_initialization(classifier=classifier, image=image, label=label,
+                                     targeted=targeted, epsilon=epsilon)
+        # classifier and image cant be set as descriptors
+        self.add_descriptors(label=label, targeted=targeted, epsilon=epsilon)
+
+    @classmethod
+    def from_testbed(
+            cls,
+            name: str,
+            label: int = 0,
+            targeted: bool = False,
+            epsilon: float = 0.05
+    ) -> "ImageAdversarial":
+        if name == "test":
```

Comment:
elif name == "resnes50":
    classifier = ResNet50()

(and move the definition of ResNet50 to the present file!)